pkg/engine/handlers/validation/validate_assert.go

```diff
@@ -122,3 +145,56 @@ func (h validateAssertHandler) Process(
 		engineapi.RulePass(rule.Name, engineapi.Validation, msg),
 	)
 }
+
+func validateOldObject(ctx context.Context, policyContext engineapi.PolicyContext, rule kyvernov1.Rule, payload map[string]any, bindings binding.Bindings) (field.ErrorList, error) {
+	if policyContext.Operation() != kyvernov1.Update {
+		return nil, nil
+	}
+
+	oldResource := policyContext.NewResource()
+
+	if rule.MatchResources.All != nil || rule.MatchResources.Any != nil {
+		matched := match.CheckMatchesResources(
+			policyContext.OldResource(),
+			kyvernov2beta1.MatchResources{
+				Any: rule.MatchResources.Any,
+				All: rule.MatchResources.All,
+			},
+			make(map[string]string),
+			kyvernov2.RequestInfo{},
+			oldResource.GroupVersionKind(),
+			"",
+		)
+		if matched != nil {
+			return nil, nil
+		}
+	}
+
+	if rule.ExcludeResources.All != nil || rule.ExcludeResources.Any != nil {
+		excluded := match.CheckMatchesResources(
+			policyContext.OldResource(),
+			kyvernov2beta1.MatchResources{
+				Any: rule.ExcludeResources.Any,
+				All: rule.ExcludeResources.All,
+			},
+			make(map[string]string),
+			kyvernov2.RequestInfo{},
+			oldResource.GroupVersionKind(),
+			"",
+		)
+		if excluded == nil {
+			return nil, nil
+		}
+	}
+
+	payload["object"] = policyContext.OldResource().Object
+	payload["oldObject"] = nil
+	payload["operation"] = kyvernov1.Create
+
+	asserttion := assert.Parse(ctx, rule.Validation.Assert.Value)
+	errs, err := assert.Assert(ctx, nil, asserttion, payload, bindings)
```

Comment:
What's the first returned `errs` and how do we react to this `errs`?

Reply:
First `errs` is violations in the payload when compared to the assertions